flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecMLPredictTableFunction.java

```diff
@@ -301,9 +337,9 @@ private Transformation<RowData> createAsyncModelPredict(
                             createTransformationMeta(ML_PREDICT_TRANSFORMATION, config),
                             new AsyncWaitOperatorFactory<>(
                                     asyncFunc,
-                                    asyncLookupOptions.asyncTimeout,
-                                    asyncLookupOptions.asyncBufferCapacity,
-                                    asyncLookupOptions.asyncOutputMode),
+                                    asyncOptions.asyncTimeout,
```

Comment:
Why don't we need these data shuffling logic in `createModelPredict`? Looks in lookup exec node, it's in sync transformation creation

Reply:
I think we can add this later if users require this. UpsertMaterialize is a complicated optimization and works if the pk of the source are different from the sink pk. What's more, it requires users use the output of the model as the part of pk.

Reply:
Regardless of upsertMaterialize, looks we have upsert key shuffling in async ordered mode, should it also be in sync mode?

Reply:
Nope. 

1. If upsert keys are subset of the the lookup keys,  we don't need to introduce shuffle here. Because planner promises the data with same upsert keys is located at the same subtask.
2. If the downstream operator doesn't use output of the predict function as upsert keys, we don't need deterministic here.

Reply:
After discussion with @lihaosky offline, I think it's a mistake to introduce shuffle for async mode. We should only introduce shuffle if we use upsert materalize.